packages/sdk/test/client/CIS3Contract.test.ts

```diff
@@ -0,0 +1,121 @@
+import { serializeCIS2Transfers } from '../../src/cis2/util.ts';
+import { serializeCIS3PermitMessage } from '../../src/cis3/util.ts';
+import {
+    ContractAddress,
+    AccountAddress,
+    Timestamp,
+    CIS3Contract,
+    CIS3,
+    BlockHash,
+    EntrypointName,
+    Parameter,
+    buildAccountSigner,
+    Energy,
+    serializeTypeValue,
+    signMessage,
+} from '../../src/index.js';
+import { getNodeClientV2 } from './testHelpers.ts';
+
+// Sponsoree and contract owner
+const SPONSOREE = AccountAddress.fromBase58(
+    '4NgCvVSCuCyHkALqbAnSX3QEC7zrfoZbig7X3ePMpk8iLod6Yj'
+);
+const SPONSOR = AccountAddress.fromBase58(
+    '3CnwP3DcvYdzHsjpU1Xme2sUS3ATK5qk2W7gi945vierMfLT7W'
+);
+const SPONSOREE_KEY =
+    '6fb89ee03ea03d038ba0bfa62dbc29eec6f30b154f797cfa5c296beb1c178428';
+// Contract that supports CIS2 tokens and the use of `permit` for transfers. SPONSOREE has a lot of tokens.
+const CIS2_MULTI_CONTRACT = ContractAddress.create(8657);
+const TOKEN_ID = '01';
+const TEST_BLOCK = BlockHash.fromHexString(
+    '5f11cac7e4b81784dda33d46ffdef94e886db3321fb3a995f6b63eef8d4b68b7'
+);
+
+const in5Minutes = () =>
+    Timestamp.fromDate(new Date(Date.now() + 1000 * 60 * 5));
```

Comment:
**Suggestion:** Add this as a `futureMinutes` helper in the `Timestamp` module.

We already have it for `TransactionExpiry`.